sdks/python/apache_beam/io/gcp/dicomio_integration_test.py

```diff
@@ -0,0 +1,215 @@
+HEALTHCARE_BASE_URL = 'https://healthcare.googleapis.com/v1'
+GCS_BASE_URL = 'https://storage.googleapis.com/storage/v1'
+PERSISTENT_DICOM_STORE_NAME = "dicom_it_persistent_store"
+TEMP_BUCKET_NAME = 'temp-storage-for-dicom-io-tests'
+TEMP_FILES_PATH = 'gs://' + TEMP_BUCKET_NAME
+META_DATA_ALL_NAME = 'Dicom_io_it_test_data.json'
+META_DATA_REFINED_NAME = 'Dicom_io_it_test_refined_data.json'
+NUM_INSTANCE = 18
+RAND_LEN = 15
+
+
+def random_string_generator(length):
+  letters_and_digits = string.ascii_letters + string.digits
+  result = ''.join((random.choice(letters_and_digits) for i in range(length)))
+  return result
+
+
+def create_dicom_store(project_id, dataset_id, region, dicom_store_id):
+  # Create a an empty DICOM store
+  credential, _ = default()
+  session = requests.AuthorizedSession(credential)
+  api_endpoint = "{}/projects/{}/locations/{}".format(
+      HEALTHCARE_BASE_URL, project_id, region)
+
+  # base of dicomweb path.
+  dicomweb_path = "{}/datasets/{}/dicomStores".format(api_endpoint, dataset_id)
+
+  response = session.post(
+      dicomweb_path, params={"dicomStoreId": dicom_store_id})
+  response.raise_for_status()
+  return response.status_code
+
+
+def delete_dicom_store(project_id, dataset_id, region, dicom_store_id):
+  # Delete an existing DICOM store
+  credential, _ = default()
+  session = requests.AuthorizedSession(credential)
+  api_endpoint = "{}/projects/{}/locations/{}".format(
+      HEALTHCARE_BASE_URL, project_id, region)
+
+  # base of dicomweb path.
+  dicomweb_path = "{}/datasets/{}/dicomStores/{}".format(
+      api_endpoint, dataset_id, dicom_store_id)
+
+  response = session.delete(dicomweb_path)
+  response.raise_for_status()
+  return response.status_code
+
+
+def get_gcs_file_http(file_name):
+  # Get gcs file from REST Api
+  api_endpoint = "{}/b/{}/o/{}?alt=media".format(
+      GCS_BASE_URL, TEMP_BUCKET_NAME, file_name)
+
+  credential, _ = default()
+  session = requests.AuthorizedSession(credential)
+
+  response = session.get(api_endpoint)
+  response.raise_for_status()
+  return response.json()
+
+
+@unittest.skipIf(DicomSearch is None, 'GCP dependencies are not installed')
+class DICOMIoIntegrationTest(unittest.TestCase):
+  def setUp(self):
+    self.test_pipeline = TestPipeline(is_integration_test=True)
+    self.project = self.test_pipeline.get_option('project')
+    self.expected_output_all_metadata = get_gcs_file_http(META_DATA_ALL_NAME)
+    self.expected_output_refined_metadata = get_gcs_file_http(
+        META_DATA_REFINED_NAME)
+
+    # create a temp Dicom store based on the time stamp
+    self.temp_dicom_store = "DICOM_store_" + random_string_generator(RAND_LEN)
+    create_dicom_store(self.project, DATA_SET_ID, REGION, self.temp_dicom_store)
+
+  def tearDown(self):
+    # clean up the temp Dicom store
+    delete_dicom_store(self.project, DATA_SET_ID, REGION, self.temp_dicom_store)
+
+  @attr('IT')
+  def test_dicom_search_all_instances(self):
```

Comment:
as you realized, each test takes 5 minutes to run. With that in mind, would it be possible to consolidate the all_instance and refined_instance searches into a single test case?

Something roughly like this:
```
with p:
  refined = p | searchrefined
  all = p | searchall
  assert_that(exp_ref, refined)
  assert_that(exp_all, all, "allmatch")
```

Reply:
pipeline reduced